EIPS/eip-7718.md

```diff
@@ -0,0 +1,418 @@
+
+Unsupported messages **SHOULD** receive a `TALKRESP` message with an empty payload.
+
+### Protocol identifiers
+
+All protocol identifiers consist of two bytes. The first byte is "`P`" (`0x50`), to indicate "the Portal network", the second byte is a specific network identifier.
+
+#### Mainnet identifiers
+
+Currently defined mainnet protocol identifiers:
+
+- Inclusive range of `0x5000` - `0x5009`: Reserved for future networks or network upgrades
+- `0x500A`: Execution State Network
+- `0x500B`: Execution History Network
+- `0x500C`: Beacon Chain Network
+- `0x500D`: Execution Canonical Transaction Index Network
+- `0x500E`: Execution Verkle State Network
+- `0x500F`: Execution Transaction Gossip Network
+
+#### Angelfood identifiers
+
+Currently defined `angelfood` protocol identifiers:
+
+- `0x504A`: Execution State Network
+- `0x504B`: Execution History Network
+- `0x504C`: Beacon Chain Network
+- `0x504D`: Execution Canonical Transaction Index Network
+- `0x504E`: Execution Verkle State Network
+- `0x504F`: Execution Transaction Gossip Network
+
+### Content Keys and Content IDs
+
+Content keys are used to request or offer specific content data. As such the content key and content data can be represented as a key:value pair.
+
+Content keys are passed as byte strings to the messages defined in the Portal wire protocol. How they are encoded is defined per content network specification.
+
+Content IDs are derived from the content keys and are used to identify where the content is located in the network. The derivation is defined per content network specification.
+
+#### SHA256 Content ID Derivation Function
+
+The SHA256 Content ID derivation function is defined as:
+
+```
+content_id = sha256(content_key)
+```
+
+### Nodes and Node IDs
+
+Nodes in the portal network are represented by their [EIP-778 Ethereum Node Record (ENR)](./eip-778.md) from the Discovery v5 network. A node's `node-id` is derived according to the node's identity scheme, which is specified in the node's ENR. A node's `node-id` represents its address in the DHT.  Node IDs are interchangeable between 32 byte identifiers and 256 bit integers.
+
+### Request - Response Messages
+
+The messages in the protocol are transmitted using the `TALKREQ` and `TALKRESP` messages from the base [Node Discovery Protocol](https://github.com/ethereum/devp2p/blob/56a498ee34ee0fb69ffd33dda026d632af4c4048/discv5/discv5-wire.md#talkreq-request-0x05).
+
+All messages in the protocol have a request-response interaction:
+
+- Request messages **MUST** be sent using a `TALKREQ` message.
+- Response messages **MUST** be sent using the corresponding `TALKRESP` message.
+
+All messages are encoded as an [SSZ Union](https://github.com/ethereum/consensus-specs/blob/04f5ec595d78c0e3e43794fb7644c18f2584770d/ssz/simple-serialize.md#union) type.
+
+```
+message = Union[ping, pong, find_nodes, nodes, find_content, content, offer, accept]
+serialized_message = serialize(message)
```

Comment:
In some other places in our specs I think we have been using this format to make it clear that this is SSZ serialization.

```suggestion
serialized_message = SSZ.serialize(message)
```